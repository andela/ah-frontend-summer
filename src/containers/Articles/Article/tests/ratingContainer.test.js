import React from 'react';
import { shallow } from 'enzyme';

import {
    RatingContainer,
    mapDispatchToProps,
    mapStateToProps
} from '../RatingContainer';

describe('rate container', () => {
    const mockRateUpdateFetch = jest.fn();
    const dispatch = jest.fn();

    const ratingProps = {
        rate: mockRateUpdateFetch,
        match: {
            params: {
                slug: "12de-de"
            }
        }
    };

    const ratingContainer = shallow(<RatingContainer {...ratingProps} />);
    const ratingInstance = ratingContainer.instance();

    afterEach(() => {
        ratingContainer.setState({ rateScore: 0 });
    });
    it("should render rating component correctly", () => {
        expect(ratingContainer).toMatchSnapshot();
    });

    it("should prompt onStarClick updates rate score in state", () => {
        ratingInstance.onStarClick(4);
        expect(mockRateUpdateFetch).toHaveBeenCalled();
        expect(ratingContainer.state().rateScore).toBe(4);
    });

    it("should map dispatch to props", () => {
        mapDispatchToProps(dispatch).rate();
        expect(dispatch.mock.calls.length).toBe(1);
    });

    it("should map state to props", () => {
        const article = { "averageRating": 5 };
        expect(mapStateToProps({ article })).toEqual({ "averageRating": 5 });
    });
});
