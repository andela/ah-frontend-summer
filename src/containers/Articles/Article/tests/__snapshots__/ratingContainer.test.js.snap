// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rate container should render rating component correctly 1`] = `
<div>
  <Rating
    averageRating={0}
    onStarClick={[Function]}
    rateScore={0}
  />
</div>
`;
